### ISSUE TITLE
Better UI For Server Selection

### ISSUE THREAD

**Issue by joelbduncan** [Slethen/ProtonVPN-Gtk#11]
Server selection UI could do with an update.

Limiting the max height and adding search functionally would clean up the visuals
and make server selection quicker

**Comment by MitchellRandom** [comment]
Yeah that would be solid, a checkbox/radio button filter ability could compliment it nicely for trying to find secure_core, p2p, Tor, or normal servers.

**Comment by joelbduncan** [comment]
I like that idea! I think I want to get this Threading issue in the bag before pushing with this though.

**Comment by Rafficer** [comment]
Or make 2 dropdowns, one by country and the other one by servers in that country.

**Comment by joelbduncan** [comment]
I like this! I think I'd like to implement that with radio buttons as suggested by @MitchellHaleUK.

I might see if I can get a mock-up tonight and maybe functionality if I can.

Good shout! Thanks for the feedback

**Comment by joelbduncan** [comment]
Work started on toggle switches very rough implementation that does work but is some ugly code!
https://github.com/Slethen/ProtonVPN-Gtk/tree/radio-btn-filter

When this is done I'll see about two Dropdowns as suggested by @Rafficer

**Comment by MitchellRandom** [comment]
I can see where you are coming from the server list building, however, the Secure Core servers will have to be approached in a different way as '-' is present in all US servers and FREE servers. Perhaps it might prove beneficial to add an additional column to indicate server types.

**Comment by joelbduncan** [comment]
That's what I was thinking, I'm also trying to clean it up so the function to write the list isn't repeated in the program.

At the moment I have this, it doesn't work unfortunately python doesn't agree with what I want to do.

![funtion](https://user-images.githubusercontent.com/1405445/38277433-4a3d40fe-3790-11e8-8abd-a47832ccb589.png)

**Comment by joelbduncan** [comment]
Toggle switches now in master.

Standard, Secure Core, Tor & P2P are selectable.

I didn't add one for free as free users can't see premium server's and I can't imagine paid users will be searching for free servers often.
